blocks/form/form.js

```diff
@@ -0,0 +1,182 @@
+function createSelect(fd) {
+  const select = document.createElement('select');
+  select.id = fd.Field;
+  if (fd.Placeholder) {
+    const ph = document.createElement('option');
+    ph.textContent = fd.Placeholder;
+    ph.setAttribute('selected', '');
+    ph.setAttribute('disabled', '');
+    select.append(ph);
+  }
+  fd.Options.split(',').forEach((o) => {
+    const option = document.createElement('option');
+    option.textContent = o.trim();
+    option.value = o.trim();
+    select.append(option);
+  });
+  if (fd.Mandatory === 'x') {
+    select.setAttribute('required', 'required');
+  }
+  return select;
+}
+
+function constructPayload(form) {
+  const payload = {};
+  [...form.elements].forEach((fe) => {
+    if (fe.type === 'checkbox') {
+      if (fe.checked) payload[fe.id] = fe.value;
+    } else if (fe.id) {
+      payload[fe.id] = fe.value;
+    }
+  });
+  return payload;
+}
+
+async function submitForm(form) {
+  const payload = constructPayload(form);
+  const resp = await fetch(form.dataset.action, {
+    method: 'POST',
+    cache: 'no-cache',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({ data: payload }),
+  });
+  await resp.text();
+  return payload;
+}
+
+function createButton(fd) {
+  const button = document.createElement('button');
+  button.textContent = fd.Label;
+  button.classList.add('button');
+  if (fd.Type === 'submit') {
+    button.addEventListener('click', async (event) => {
+      const form = button.closest('form');
+      if (form.checkValidity()) {
+        event.preventDefault();
+        button.setAttribute('disabled', '');
+        await submitForm(form);
+        const redirectTo = fd.Extra;
+        window.location.href = redirectTo;
+      }
+    });
```

Comment:
Two issues with this:
1. the validity check should not happen upon button click, but upon form submission (there are other ways to submit a form that would not get detected here)
2. disabling the submit button can be a good idea, but should happen in the validation cade

Alternative to the validity check upon form submission, there could be a check upon entry, but it's important to get this right:
- if you validity check upon first entry, all required fields will be marked as invalid, which is annoying (let me get to them first)
- if you validity check upon entry, email fields will first be marked invalid, until you get to the domain part, so not ideal either

Reply:
I will update the main form block in the helix-website then

Reply:
Update it here, and we merge `helix-website` later.

Reply:
This code has already been removed as per Satya's review comments since nedbank has no use case of submit.

Reply:
There are multiple patterns related to validation

1. validate on as you type (which is there in nedbank as well), but it has problems when there are multiple constraints
2. validate after the change (or blur) event (this works correctly) but should happen only if there is a change in the value. Just using tabs to iterate through the fields should not validate.
3. validate on submit (the submit event) and on click in the case when we are not using form submit (which is happening here; I am assuming form service doesn't accept `application/url-encoded` content type). 
Both the click and submit event have the same problem as `form.submit` API doesn't trigger any event.

> disabling the submit button can be a good idea, but should happen in the validation cade
@trieloff I did not get this part. disabling is required in this case since we are doing a `fetch` here which is asynchronous and can lead to double submit. It should happen just before submit, validation is synchronous.

Reply:
I have resolved the first part of it.